src/Database/Type/IntegerType.php

```diff
@@ -113,7 +113,7 @@ public function toStatement(mixed $value, Driver $driver): int
      */
     public function marshal(mixed $value): ?int
     {
-        if ($value === null || $value === '' || !is_numeric($value)) {
+        if ($value === '' || !is_numeric($value)) {
```

Comment:
Are we sure it can never be null, even through some API call?
Then this would be a possible regression